File: r/p7.r
#!/usr/bin/Rscript

printf <- function(...) invisible(print(sprintf(...)))

IsPrime <- function(n){
  if(n == 1){
    return(F)
  }else if(n < 4){
    return(T)
  }else if(n %% 2 == 0){
    return(F)
  }else if(n < 9){
    return(T)
  }else if(n %% 3 == 0){
    return(F)
  }else{
    r  <- floor(sqrt(n))
    f  <- 5
    while (f <= r){
      if (n %% f == 0){
        return(F)
      }else if(n %% (f  +  2) == 0){
        return(F)
      }
      f  <- f + 6
    }
  }
  return(T)
}

limit  <- 10001
args  <- commandArgs(TRUE)
if(length(args) > 0){
  limit  <- as.integer(args[1])
}
count  <-  1
val  <-  1
while(count < limit){
  val  <- val + 2
  if (IsPrime(val)){
    count  <- count + 1
  }
}

printf("%i", val)

File: fsharp/p7.fs
let IsPrime n : bool = 
  if n = 1 then
    false
  elif n < 4 then
    true
  elif n % 2 = 0 then
    false
  elif n < 9 then
    true
  elif n % 3 = 0 then
    false
  else
    let r = sqrt (n |> float) |> int
    let mutable f = 5
    let mutable ret = true
    while f <= r && ret = true do
      if n % f = 0 then
        ret <- false
      elif (n % (f+2)) = 0 then
        ret <- false
      f <- f + 6
    ret

[<EntryPoint>]
let main(argv) = 
  let limit = if (Array.length argv > 0) then argv.[0] |> int
              else 10001
  let mutable count = 1
  let mutable v = 1
  while count < limit do
    v <- v + 2
    if IsPrime v then
      count <- count + 1
  printfn "%i" v
  0
